streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignorTest.java

```diff
@@ -1953,6 +1975,54 @@ public void shouldRequestCommittedOffsetsForPreexistingSourceChangelogs() {
         EasyMock.verify(consumerClient);
     }
 
+    @Test(timeout = 30 * 1000)
+    public void shouldCompleteLargeAssignmentInAReasonableAmountOfTime() {
+        builder.addSource(null, "source", null, null, null, TOPICS_LIST_XL.toArray(new String[0]));
+        builder.addProcessor("processor", new MockProcessorSupplier(), "source");
+        builder.addStateStore(new MockKeyValueStoreBuilder("store", false), "processor");
+
+        for (int i = 0; i < NUM_CONSUMERS_XL; ++i) {
+            subscriptions.put("consumer-" + i,
+                              new Subscription(
+                                  TOPICS_LIST_XL,
+                                  getInfo(uuidForInt(i), EMPTY_TASKS, EMPTY_TASKS).encode())
+            );
+        }
+        createMockTaskManager(EMPTY_TASKS, EMPTY_TASKS);
+        createMockAdminClient(CHANGELOG_END_OFFSETS_XL);
+        configurePartitionAssignorWith(singletonMap(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, 3));
+
+        final Map<String, Assignment> assignments =
+            partitionAssignor.assign(CLUSTER_METADATA_XL, new GroupSubscription(subscriptions)).groupAssignment();
+
+        // Use the assignment to generate the subscriptions' prev task data for the next rebalance
+        for (int i = 0; i < NUM_CONSUMERS_XL; ++i) {
+            final String consumer = "consumer-" + i;
+            final Assignment assignment = assignments.get(consumer);
+            final AssignmentInfo info = AssignmentInfo.decode(assignment.userData());
+
+            subscriptions.put("consumer-" + i,
+                              new Subscription(
+                                  TOPICS_LIST_XL,
+                                  getInfo(uuidForInt(i), new HashSet<>(info.activeTasks()), info.standbyTasks().keySet()).encode(),
+                                  assignment.partitions())
+            );
+        }
+
+        final Map<String, Assignment> secondAssignments =
```

Comment:
req: Could you please remove variable `secondAssignments` since it is never used?